src/back/space/src/service/service.cpp

```diff
@@ -0,0 +1,459 @@
+#include "userver/components/component_config.hpp"
+#include "userver/components/component_context.hpp"
+#include "userver/formats/json/serialize.hpp"
+#include "userver/http/url.hpp"
+#include <userver/crypto/base64.hpp>
+
+namespace svetit::space {
+
+/*static*/ yaml_config::Schema Service::GetStaticConfigSchema()
+{
+	return yaml_config::MergeSchemas<components::LoggableComponentBase>(R"(
+type: object
+description: Main service component
+additionalProperties: false
+properties:
+  can-create:
+    type: string
+    description: Can create new spaces
+  default-space:
+    type: string
+    description: Default space to join
+  spaces-limit-for-user:
+    type: string
+    description: How many spaces user can join
+  items-limit-for-list:
+    type: string
+    description: How many items list may contain
+)");
+}
+
+Service::Service(
+		const components::ComponentConfig& conf,
+		const components::ComponentContext& ctx)
+	: components::LoggableComponentBase{conf, ctx}
+	, _repo{ctx.FindComponent<Repository>()}
+	, _canCreate{conf["can-create"].As<bool>()}
+	, _defaultSpace{conf["default-space"].As<std::string>()}
+	, _spacesLimitForUser{conf["spaces-limit-for-user"].As<int>()}
+	, _itemsLimitForList{conf["items-limit-for-list"].As<int>()}
+{}
+
+std::vector<model::Space> Service::GetList(const std::string userId, unsigned int start, unsigned int limit)
+{
+	return _repo.Space().SelectByUserId(utils::BoostUuidFromString(userId), start, limit);
+}
+
+std::vector<model::Space> Service::GetAvailableList(const std::string userId, unsigned int start, unsigned int limit)
+{
+	return _repo.Space().SelectAvailable(utils::BoostUuidFromString(userId), start, limit);
+}
+
+int Service::GetCount(const std::string userId) {
+	return _repo.Space().CountByUserId(utils::BoostUuidFromString(userId));
+}
+
+int Service::GetAvailableCount(const std::string userId) {
+	return _repo.Space().CountAvailable(utils::BoostUuidFromString(userId));
+}
+
+std::vector<model::SpaceInvitation> Service::GetInvitationList(unsigned int start, unsigned int limit)
+{
+	return _repo.SpaceInvitation().Select(start,limit);
+}
+
+int Service::GetInvitationsCount() {
+	return _repo.SpaceInvitation().Count();
+}
+
+std::vector<model::SpaceLink> Service::GetLinkList(unsigned int start, unsigned int limit)
+{
+	return _repo.SpaceLink().Select(start,limit);
+}
+
+int Service::GetLinksCount() {
+	return _repo.SpaceLink().Count();
+}
+
+std::vector<model::SpaceUser> Service::GetUserList()
+{
+	return {};
+}
+
+int Service::GetUsersCount() {
+	return 0;
+}
+
+bool Service::isSpaceExistsByKey(std::string key) {
+	return _repo.Space().IsExists(key);
+}
+
+bool Service::isCanCreate() {
+	return _canCreate;
+}
+
+int Service::CountInvitationAvailable(const boost::uuids::uuid currentUserId) {
+	return _repo.SpaceInvitation().GetAvailableCount(currentUserId);
+}
+
+bool Service::CheckKeyByRegex(std::string key) {
+	std::string regex = "[a-z0-9_]*";
+	std::regex rx(regex);
```

Comment:
static - тут означает что этот объект будет создан один раз при первом вызове и затем при последующих вызовах будет использоваться повторно. Для regex это важно потому что копиляция из шаблона это ресурсозатратная операция.
```suggestion
	static const std::regex rx("[a-z0-9_]*");
```

Reply:
Понял. Готово.